### PR TITLE
OrgRoleContainer

### PR DESCRIPTION
**Pull request by UnsungHero97**
depends on:

https://github.com/openlattice/lattice-utils/pull/133
https://github.com/openlattice/lattice-js/pull/1003
https://github.com/openlattice/lattice-sagas/pull/623
https://github.com/openlattice/lattice-sagas/pull/624

Screen Shot 2020-09-22 at 3 38 10 PM

Screen Shot 2020-09-22 at 4 40 26 PM

### CODE REVIEW

File: src/containers/org/components/DataSetPermissionsCard.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  dataSet,
+  principal,
+} :{
+  dataSet :EntitySet;
+  principal :Principal;
+}) => {
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [targetPermissionType, setTargetPermissionType] = useState();
+
```
Comment (anzioka) [inline src/containers/org/components/DataSetPermissionsCard.js:76]: missing initial value argument to `useState`

File: src/containers/org/roles/OrgRoleContainer.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          {
+            reducedRS === RequestStates.SUCCESS && (
+              <StackGrid>
+                {
+                  permissionsCount > MAX_PER_PAGE && (
+                    <PaginationToolbar
+                        page={paginationPage}
+                        count={permissionsCount}
+                        onPageChange={handleOnPageChange}
+                        rowsPerPage={MAX_PER_PAGE} />
```
Comment (anzioka) [inline src/containers/org/roles/OrgRoleContainer.js:142]: sort

File: src/containers/org/components/DataSetPermissionsCard.js (hunk 1 above)
Comment (anzioka) [reply src/containers/org/components/DataSetPermissionsCard.js:76]: though I suppose the implicit value is `undefined`

File: src/containers/org/roles/OrgRoleContainer.js (hunk 2 above)
Comment (UnsungHero97) [reply src/containers/org/roles/OrgRoleContainer.js]: fixed

File: src/containers/org/roles/OrgRoleContainer.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                  )
+                }
+                {
+                  pageDataSets.map((dataSet :EntitySet) => (
+                    <DataSetPermissionsCard dataSet={dataSet} key={dataSet.id} principal={role.principal} />
+                  )).valueSeq()
```
[the remark this answers is not in the document]
Comment (UnsungHero97) [reply src/containers/org/roles/OrgRoleContainer.js:148]: Why are you surprised? `.map()` will update the values, and then `.valueSeq()` will return the values. Both operations work on the values, so the order of operations here doesn't matter.

File: src/containers/org/components/DataSetPermissionsCard.js (hunk 1 above)
Comment (UnsungHero97) [reply src/containers/org/components/DataSetPermissionsCard.js:76]: The initial value is meant to be `undefined`. We don't know the "target" until the user clicks on it.

**Review by katherinebernstein — APPROVED** [review]

File: src/core/edm/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  if (action.type === getEntityDataModelTypes.case(action.type)) {
+    return getEntityDataModelTypesReducer(state, action);
+  }
+
+  if (action.type === getEntitySet.case(action.type)) {
+    return getEntitySetReducer(state, action);
+  }
+
+  // console.log(action.type);
+  if (action.type === getEntitySets.case(action.type)) {
```
Comment (solomontang) [inline src/core/edm/reducers/index.js:62]: Rogue comment

**Review by makosblade — APPROVED** [review]

File: src/core/edm/reducers/index.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+} from '../actions';
+
+const {
+  GET_ENTITY_SET,
+  GET_ENTITY_SETS,
+  getEntitySet,
+  getEntitySets,
+} = EntitySetsApiActions;
+
+const INITIAL_STATE :Map = fromJS({
```
Comment (solomontang) [inline src/core/edm/reducers/index.js:36]: An interesting pattern. You're basically composing reducers without using `compose`. But this pattern also separates the initial state of each reducer from the actual reducer logic.

**Pull request merged**